db/db_impl/db_impl.cc

```diff
@@ -2315,6 +2326,21 @@ Status DBImpl::MultiGetImpl(
       super_version->current->MultiGet(read_options, &range, callback,
                                        is_blob_index);
     }
+    curr_value_size = range.GetValueSize();
+    if (curr_value_size > read_options.value_size_soft_limit) {
+      break;
```

Comment:
Set ```s``` to ```Status::Aborted()```? We have logic to abort the MultiGet for multiple column families around line 2096 that looks at the overall status. You just need to fix the assert in 2096 and 2255 as it currently only expects ```Status::TimedOut()```.